tests/linalg_test.py

```diff
@@ -269,6 +269,29 @@ def check_left_eigenvectors(a, w, vl):
     self._CompileAndCheck(partial(jnp.linalg.eig), args_maker,
                           rtol=1e-3)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_shape={}".format(
+           jtu.format_shape_dtype_string(shape, dtype)),
+       "shape": shape, "dtype": dtype, "rng_factory": rng_factory
+       }
+      for shape in [(4, 4), (5, 5), (8, 8), (7, 6, 6)]
+      for dtype in float_types + complex_types
+      for rng_factory in [jtu.rand_default]))
+  # TODO(phawkins): enable when there is an eigendecomposition implementation
+  # for GPU/TPU.
+  @jtu.skip_on_devices("gpu", "tpu")
+  def testEigvalsGrad(self, shape, dtype, rng_factory):
+    # This test sometimes fails for large matrices. I (@j-towns) suspect, but
+    # haven't checked, that might be because of perturbations causing the
+    # ordering of eigenvalues to change, which will trip up check_grads. So we
+    # just test on small-ish matrices.
```

Comment:
One way to solve this would be to explicitly create matrices without near degenerate eigenvalues, e.g., by constructing them via a matrix decomposition. But really, the checks you have here already seem plenty comprehensive to me.

Reply:
Yeah I agree, but I'm not going to implement this now, just thought it worth putting a note there in case someone has the time and motivation to improve the situation in the future 🙂.